apps/src/templates/studioHomepages/UiTips.jsx

```diff
@@ -81,60 +82,56 @@ const UiTips = React.createClass({
     }
 
     this.setState(newState);
-  },
+  };
 
-  afterDialogCancel() {
-    let newState = {...this.state, showingDialog: null};
-    this.setState(newState);
+  afterDialogCancel = () => {
+    this.setState({...this.state, showingDialog: null});
 
     trackEvent("ui_tips", this.props.tipId, "after_dialog_cancel");
-  },
+  };
 
-  afterDialogConfirm() {
+  afterDialogConfirm = () => {
     if (this.props.afterDialog.onConfirm.action === "url") {
       window.open(this.props.afterDialog.onConfirm.url, "_blank");
     }
-    let newState = {...this.state, showingDialog: null};
-    this.setState(newState);
+    this.setState({...this.state, showingDialog: null});
 
     trackEvent("ui_tips", this.props.tipId, "after_dialog_confirm");
-  },
+  };
 
-  beforeDialogCancel() {
-    let newState = {...newState, showInitialTips: false, showingDialog: null};
```

Comment:
I think there was a subtle bug here:
```js
let newState = {...newState};
// instead of
let newState = {...this.state};
```
We were throwing away the component state, besides the new stuff we add in this case.  In practice it didn't matter because when you cancel the "before" dialog in the one place this is used, you're saying you don't want to see any more tooltips so we'd never run into a consequence of the bad state.